idaes/generic_models/unit_models/distillation/reboiler.py

```diff
@@ -545,40 +560,107 @@ def rule_bottoms_enth(self, t):
+
+            for k in state_dict.keys():
+                if state_dict[k].is_indexed():
+                    state_args[k] = {}
+                    for m in state_dict[k].keys():
+                        state_args[k][m] = state_dict[k][m].value
+                else:
+                    state_args[k] = state_dict[k].value
+
         # Initialize the inlet and outlet state blocks
-        self.control_volume.initialize(outlvl=outlvl)
+        flags = self.control_volume.initialize(state_args=state_args,
+                                               solver=solver,
+                                               optarg=optarg,
+                                               outlvl=outlvl,
+                                               hold_state=True)
+
+        if not self.control_volume.properties_out[0].temperature.fixed:
+
+            # Estimating a good guess for the reboiler temperature based on
+            # a vapor fraction of 0.5, bubble and dew
+            # point. This helps with initializing the system within the two
+            # phase envelope before activating the boilup ratio constraint.
+
+            alpha = 0.5 * (
+                self.control_volume.properties_in[0].temperature_dew.value -
+                self.control_volume.properties_in[0].
+                temperature_bubble.value) + \
+                self.control_volume.properties_in[0].temperature_bubble.value
+
+            self.eq_boilup_ratio.deactivate()
+            self.control_volume.properties_out[0].temperature.fix(alpha)
+
+            if degrees_of_freedom(self) == 0:
+                with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                    res = solver.solve(self, tee=slc.tee)
+                init_log.info(
+                    "Initialization Step 1 Complete, {}.".
+                    format(idaeslog.condition(res))
+                )
+            else:
+                raise Exception("State vars fixed but degrees of freedom "
+                                "for reboiler is not zero during "
+                                "initialization.")
 
-        if solver is not None:
-            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
-                res = solver.solve(self, tee=slc.tee)
-            init_log.info(
-                "Initialization Complete, {}.".format(idaeslog.condition(res))
-            )
+            self.control_volume.properties_out[0].temperature.unfix()
+            self.eq_boilup_ratio.activate()
         else:
-            init_log.warning(
-                "Solver not provided during initialization, proceeding"
-                " with deafult solver in idaes.")
-            solver = get_default_solver()
+            # In this case, outlet temperature has already been fixed by user
+            # and boil up ratio is assumed not to be fixed, otherwise the
+            # system is over specified.
+
+            self.eq_boilup_ratio.deactivate()
```

Comment:
The logic here seems a little off. You deactivate the boilup ratio equation and expect the problem to have 0 degrees of freedom, solve it and then reactivate the boilup ratio equation. This means that either the boilup ratio equation is only used to calculate the boilup ratio (I assume this is the case) in which case why don't you just leave it active, or that the problem with the equation is not square.

Reply:
I deactivate the boil up ratio constraint which is (`V/B == boilup_ratio`) and fix the outlet reboiler temperature to a temperature within the two phase envelope. This is an easier spec to initialize than by having the constraint active and figuring out the temperature that satisfies the system of equations. Let me know if you have any other suggestions for this step. I based the temperature guess on the following:

```
alpha = 0.5 * (
                self.control_volume.properties_in[0].temperature_dew.value -
                self.control_volume.properties_in[0].
                temperature_bubble.value) + \
                self.control_volume.properties_in[0].temperature_bubble.value

            self.eq_boilup_ratio.deactivate()
            self.control_volume.properties_out[0].temperature.fix(alpha)
```

Reply:
That is true for the first part of the if statement I believe, but this is for the case when T was already fixed by the user.

In this case, I think you should just be able to solve for boilup from the beginning (i.e. not need to deactivate it).

Reply:
Okay, good question but see comment on lines 613-615. If the outlet temperature was fixed, the boil up ratio var cannot be fixed and hence, I have a check to ensure that the DOF=0.  I can do better by raising appropriate error messages when DOF is not 0. Not sure if I answered what you asked for.